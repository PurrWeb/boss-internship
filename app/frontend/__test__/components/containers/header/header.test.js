import React from 'react';
import Header from '~/components/containers/header/header.js';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

const quickMenu = JSON.parse(`[{"name":"Venue","color":"#e74c3c","items":[{"description":"Rota","permitted":true,"path":"/rotas"},{"description":"Security Rota","permitted":true,"path":"/security_rotas"},{"description":"Change Orders","permitted":true,"path":"/change_orders"},{"description":"Safe Checks","permitted":true,"path":"/safe_checks"}]},{"name":"Staff Members","color":"#27ae60","items":[{"description":"Hours Confirmation","permitted":true,"path":"/hours_confirmation/current"},{"description":"Holidays","permitted":true,"path":"/holidays?date=10-07-2017"},{"description":"Add Staff Member","permitted":true,"path":"/staff_members/new"}]},{"name":"Reports","color":"#9b59b6","items":[{"description":"Daily Report","permitted":true,"path":"/daily_reports"},{"description":"Weekly Report","permitted":true,"path":"/weekly_reports"},{"description":"Payroll Report","permitted":true,"path":"/payroll_reports"}]},{"name":"Admin: General","color":"#e67e22","items":[{"description":"Names","path":"/names"},{"description":"Venues","path":"/venues"}]},{"name":"Admin: Users","color":"#1abc9c","items":[{"description":"Invites","path":"/invites"}]},{"name":"Admin: Staff Members","color":"#3498db","items":[{"description":"Staff Type","path":"/staff_types"},{"description":"Pay Rates","path":"/pay_rates"},{"description":"Staff Vetting","path":"/staff_vetting"},{"description":"Staff Tracking","path":"/staff_tracking"}]},{"name":"Admin: Venue","color":"#c0392b","items":[{"description":"Daily Report","permitted":true,"path":"/daily_reports"},{"description":"Weekly Report","permitted":true,"path":"/weekly_reports"},{"description":"Payroll Report","permitted":true,"path":"/payroll_reports"}]},{"name":"Admin: Reports","color":"#f39c12","items":[{"description":"Finance Report","path":"/finance_reports"},{"description":"Yearly Report","path":"/yearly_reports"}]}]`);
const user = {
  id: 1,
  name: 'Test User'
};

const component = shallow(
  <Header quickMenu={quickMenu} user={user} />
);

describe('Quick menu', () => {
  test('should opens, when click on icon', () => {
    expect(component).toMatchSnapshot();
    component.find('.boss-page-header__action_role_search').simulate('click');
    expect(component).toMatchSnapshot();
  })
})
