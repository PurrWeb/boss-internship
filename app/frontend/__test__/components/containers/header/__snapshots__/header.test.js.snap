// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quick menu should opens, when click on icon 1`] = `
<header
  className="boss-page-header"
>
  <div
    className="boss-page-header__inner"
  >
    <div
      className="boss-page-header__group boss-page-header__group_role_logo"
    >
      <a
        className="boss-page-header__logo"
        href="/"
      >
        Boss
      </a>
    </div>
    <button
      className="boss-page-header__action boss-page-header__action_role_search"
      onClick={[Function]}
    >
      Search
    </button>
    <button
      className="boss-page-header__action boss-page-header__action_role_profile"
      onClick={[Function]}
    >
      Profile
    </button>
  </div>
</header>
`;

exports[`Quick menu should opens, when click on icon 2`] = `
<header
  className="boss-page-header"
>
  <div
    className="boss-page-header__inner"
  >
    <div
      className="boss-page-header__group boss-page-header__group_role_logo"
    >
      <a
        className="boss-page-header__logo"
        href="/"
      >
        Boss
      </a>
    </div>
    <button
      className="boss-page-header__action boss-page-header__action_role_search"
      onClick={[Function]}
    >
      Search
    </button>
    <button
      className="boss-page-header__action boss-page-header__action_role_profile"
      onClick={[Function]}
    >
      Profile
    </button>
    <HeaderDropdown
      closeDropdown={[Function]}
      handleEscPress={[Function]}
      quickMenu={
        Array [
          Object {
            "color": "#e74c3c",
            "items": Array [
              Object {
                "description": "Rota",
                "path": "/rotas",
                "permitted": true,
              },
              Object {
                "description": "Security Rota",
                "path": "/security_rotas",
                "permitted": true,
              },
              Object {
                "description": "Change Orders",
                "path": "/change_orders",
                "permitted": true,
              },
              Object {
                "description": "Safe Checks",
                "path": "/safe_checks",
                "permitted": true,
              },
            ],
            "name": "Venue",
          },
          Object {
            "color": "#27ae60",
            "items": Array [
              Object {
                "description": "Hours Confirmation",
                "path": "/hours_confirmation/current",
                "permitted": true,
              },
              Object {
                "description": "Holidays",
                "path": "/holidays?date=10-07-2017",
                "permitted": true,
              },
              Object {
                "description": "Add Staff Member",
                "path": "/staff_members/new",
                "permitted": true,
              },
            ],
            "name": "Staff Members",
          },
          Object {
            "color": "#9b59b6",
            "items": Array [
              Object {
                "description": "Daily Report",
                "path": "/daily_reports",
                "permitted": true,
              },
              Object {
                "description": "Weekly Report",
                "path": "/weekly_reports",
                "permitted": true,
              },
              Object {
                "description": "Payroll Report",
                "path": "/payroll_reports",
                "permitted": true,
              },
            ],
            "name": "Reports",
          },
          Object {
            "color": "#e67e22",
            "items": Array [
              Object {
                "description": "Names",
                "path": "/names",
              },
              Object {
                "description": "Venues",
                "path": "/venues",
              },
            ],
            "name": "Admin: General",
          },
          Object {
            "color": "#1abc9c",
            "items": Array [
              Object {
                "description": "Invites",
                "path": "/invites",
              },
            ],
            "name": "Admin: Users",
          },
          Object {
            "color": "#3498db",
            "items": Array [
              Object {
                "description": "Staff Type",
                "path": "/staff_types",
              },
              Object {
                "description": "Pay Rates",
                "path": "/pay_rates",
              },
              Object {
                "description": "Staff Vetting",
                "path": "/staff_vetting",
              },
              Object {
                "description": "Staff Tracking",
                "path": "/staff_tracking",
              },
            ],
            "name": "Admin: Staff Members",
          },
          Object {
            "color": "#c0392b",
            "items": Array [
              Object {
                "description": "Daily Report",
                "path": "/daily_reports",
                "permitted": true,
              },
              Object {
                "description": "Weekly Report",
                "path": "/weekly_reports",
                "permitted": true,
              },
              Object {
                "description": "Payroll Report",
                "path": "/payroll_reports",
                "permitted": true,
              },
            ],
            "name": "Admin: Venue",
          },
          Object {
            "color": "#f39c12",
            "items": Array [
              Object {
                "description": "Finance Report",
                "path": "/finance_reports",
              },
              Object {
                "description": "Yearly Report",
                "path": "/yearly_reports",
              },
            ],
            "name": "Admin: Reports",
          },
        ]
      }
    />
  </div>
</header>
`;
